rpc/ethereum/backend/backend.go

```diff
@@ -374,10 +381,9 @@ func (e *EVMBackend) EthBlockFromTendermint(
 
 			tx := ethMsg.AsTransaction()
 
-			// check tx exists on EVM and it has the correct block height
-			ethTx, err := e.GetTxByEthHash(tx.Hash())
-			if err != nil || ethTx.Height != block.Height {
-				e.logger.Debug("failed to query eth tx", "hash", tx.Hash().Hex())
+			// check tx exists on EVM by cross checking with blockResults
+			if txResults[i].Code != 0 {
```

Comment:
if Code is not 0, it's EVM tx?
maybe add some comment about it

Reply:
if `Code != 0`, it's a failed transaction.